bokeh/document.py

```diff
@@ -359,6 +365,16 @@ def roots(self):
         return list(self._roots)
 
     @property
+    def session_context(self):
+        return self._session_context
+
+    @session_context.setter
+    def session_context(self, session_context):
+        if self._session_context != session_context:
+            self._session_context = session_context
+            self._trigger_on_change(SessionContextChangedEvent(self, session_context))
```

Comment:
It might be simpler to say that the session context can't change after document creation - then there's no need for a change event or a setter, but more importantly, no need for apps to write code to listen to the event.